gems/aws-sdk-core/lib/aws-sdk-core/assume_role_credentials.rb

```diff
@@ -39,6 +39,7 @@ class AssumeRoleCredentials
     #      end
     #
     def initialize(options = {})
+      @metrics = options.delete(:metrics)
```

Comment:
I'm not sure if this is the pattern we should do. Since refresh is called on initialization (with super) to fetch credentials, we should just wrap the refresh logic with `Aws::Plugins::UserAgent.metric(WHATEVER) do .. refresh .. end`. I assume you added this because of nested calls? Wrapping multiple `UserAgent.metric do..end` blocks will take care of that.